### ISSUE TITLE
rec: --trace leaves out prefix on DNSSEC LOG lines

### ISSUE THREAD

**Issue by Habbie** [PowerDNS/pdns#5818]
- Program: Recursor
 - Issue type: Enhancement

### Short description
Recursor `trace` output has a prefix that tells you the query/mthread ID, and gives a tree-like appearance to the trace by indenting each line. DNSSEC validation log lines do not have this prefix.

Example:
```
Oct 12 15:59:00 [1] validation state was Secure, state update is Bogus, validation state is now Bogus
Oct 12 15:59:00 [1] : got status Secure for name apexonly.nlnetlabs.nl (from apexonly.nlnetlabs.nl)
Oct 12 15:59:00 [1] : got initial zone status Secure for record apexonly.nlnetlabs.nl
Oct 12 15:59:00 [1] Validating non-additional record for apexonly.nlnetlabs.nl
Oct 12 15:59:00 [1] Retrieving DNSKeys for apexonly.nlnetlabs.nl
Oct 12 15:59:00 [1]  apexonly.nlnetlabs.nl: Wants DNSSEC processing, auth data in query for DNSKEY
Oct 12 15:59:00 [1]  apexonly.nlnetlabs.nl: Found cache hit for DNSKEY: 256 3 8 AwEAAZad+sq/6SGNySjBlWkn6nQ9NSMBV0iPbRt8Xb4wgu7RKwsq0+4zn/7H3wiBcnRIPE7E/q9z+fl2wvN3rZ3bDghquv66tQf/MK+B5s7zmI9SmaZqbxEjeVrDwuo/GrCm5cb+34eQCTel8CtOVNbngFiDaR4aB1/Bnj9Z5WRnVwvZ[ttl=3600]
Oct 12 15:59:00 [1]  apexonly.nlnetlabs.nl: updating validation state with cache content for apexonly.nlnetlabs.nl to Secure
Oct 12 15:59:00 [1] Retrieved 1 DNSKeys for apexonly.nlnetlabs.nl, state is Secure
Oct 12 15:59:00 [1] Going to validate 1 record contents with 1 sigs and 1 keys for apexonly.nlnetlabs.nl
Oct 12 15:59:00 Signature is expired (inception: 1380458760, expiration: 1382877960, now: 1507816740)
Oct 12 15:59:00 signature invalid
Oct 12 15:59:00 [1] Bogus!
Oct 12 15:59:00 [1] apexonly.nlnetlabs.nl: determining status after receiving this packet
Oct 12 15:59:00 [1] apexonly.nlnetlabs.nl: got negative caching indication for 'apexonly.nlnetlabs.nl|A'
Oct 12 15:59:00 [1] apexonly.nlnetlabs.nl: status=noerror, other types may exist, but we are done (have negative SOA) (have aa bit)
Oct 12 15:59:00 Starting validation of answer to apexonly.nlnetlabs.nl|A for 127.0.0.1:54185
Oct 12 15:59:00 Answer to apexonly.nlnetlabs.nl|A for 127.0.0.1:54185 validates as Bogus
Oct 12 15:59:00 Sending out SERVFAIL for apexonly.nlnetlabs.nl|A because recursor or query demands it for Bogus results
Oct 12 15:59:00 2 [1/1] answer to question 'apexonly.nlnetlabs.nl|A': 0 answers, 0 additional, took 23 packets, 572.668 netw ms, 646.069 tot ms, 0 throttled, 0 timeouts, 0 tcp connections, rcode=0, dnssec=Bogus
```

Note the two blocks that are lacking `[1]`.

While looking at fixing this, I noticed that all parts of the recursor that do get this right have a copy of this code:
```
  string prefix;
  if(doLog()) {
    prefix=d_prefix;
    prefix.append(depth, ' ');
  }
```

It might be nice to clean that up first.

**Comment by rgacogne** [comment]
While I agree this would be nice to have, it's probably not easy. Getting the prefix outside of the `SyncRes` class is a pain, and I don't want to move the validation code in that class.

**Comment by Habbie** [comment]
Yes, I had the same impression :(

**Comment by omoerbeek** [comment]
I have structured logging wip code that fixes this. I would not spend time on the old style logging code.

**Comment by omoerbeek** [comment]
Fixed by #12434
